### PR TITLE
Add a `DEFAULT_HELPER_MANAGER` feature flag

### PR DESCRIPTION
**Pull request by Windvis**
I gave adding a feature flag a shot based on the information in #1383. I went for a similar approach as the Ember feature flagging system (albeit a bit simpler). Let me know if I should do things differently!

Closes https://github.com/glimmerjs/glimmer-vm/issues/1383

### CODE REVIEW

File: packages/@glimmer/integration-tests/test/managers/helper-manager-test.ts [hunk 1]
```diff
@@ -10,423 +10,426 @@ import {
   TestHelperManager,
 } from '../..';
 import { Arguments, Owner } from '@glimmer/interfaces';
+import { FEATURES } from '@glimmer/global-context';
 
-class HelperManagerTest extends RenderTest {
-  static suiteName = 'Helper Managers';
+if (FEATURES.DEFAULT_HELPER_MANAGER) {
```
Comment (Windvis) [inline packages/@glimmer/integration-tests/test/managers/helper-manager-test.ts:15]: It seems I disabled too many tests here. I'll try to limit it to the needed tests tomorrow.

File: packages/@glimmer/global-context/index.ts [hunk 2]
```diff
@@ -179,6 +189,8 @@ export default function setGlobalContext(context: GlobalContext) {
   warnIfStyleNotTrusted = context.warnIfStyleNotTrusted;
   assert = context.assert;
   deprecate = context.deprecate;
+
+  assign(FEATURES, context.FEATURES);
```
Comment (Windvis) [inline packages/@glimmer/global-context/index.ts:193]: I ran into the Object.assign error that was mentioned here as well: https://github.com/glimmerjs/glimmer-vm/pull/1338#issuecomment-924446617 

Do we still support IE11 here or can we just [remove the code that triggers the error](https://github.com/glimmerjs/glimmer-vm/blob/4f1bef0d9a8a3c3ebd934c5b6e09de4c5f6e4468/test/index.html#L42-L46)?

File: packages/@glimmer/global-context/index.ts [hunk 3]
```diff
@@ -14,6 +14,13 @@
  */
 
 import { DEBUG } from '@glimmer/env';
+import { assign } from '@glimmer/util';
+
+//////////
+
+export const FEATURES = {
```
Comment (Windvis) [inline packages/@glimmer/global-context/index.ts:21]: I just placed it near the top of the file, let me know if I should put it somewhere else.

File: packages/@glimmer/global-context/index.ts [hunk 4]
```diff
@@ -14,6 +14,13 @@
...
+  DEFAULT_HELPER_MANAGER: false,
+};
```
Comment (chancancode) [inline packages/@glimmer/global-context/index.ts:23]: Let's make this a top level `let` variable:

```suggestion
export let FEATURE_DEFAULT_HELPER_MANAGER = false;
```

The `const` here only mean the object cannot be reassigned, but the POJO can be mutated by whoever imports it, which introduces a "backdoor" for changing the value of the flags without going through the `setGlobalContext` API. (Using an object in that API is fine though.)

File: packages/@glimmer/global-context/index.ts (hunk 2 above)
Comment (chancancode) [reply packages/@glimmer/global-context/index.ts:193]: We don't need to support IE11, but anyway with the change proposed above this is no longer relevant.

File: packages/@glimmer/global-context/package.json [hunk 5]
```diff
@@ -3,7 +3,8 @@
   "version": "0.84.0",
   "repository": "https://github.com/glimmerjs/glimmer-vm/tree/master/packages/@glimmer/global-context",
   "dependencies": {
-    "@glimmer/env": "^0.1.7"
+    "@glimmer/env": "^0.1.7",
+    "@glimmer/util": "0.84.0"
```
Comment (chancancode) [inline packages/@glimmer/global-context/package.json:7]: Can be removed now (see above)

File: packages/@glimmer/global-context/index.ts (hunk 4 above)
Comment (chancancode) [reply packages/@glimmer/global-context/index.ts:23]: Also let's default this to `true` by default.

File: packages/@glimmer/integration-tests/lib/modes/env.ts [hunk 6]
```diff
@@ -148,6 +148,10 @@ setGlobalContext({
       actualDeprecations.push(msg);
     }
   },
+
+  FEATURES: {
+    DEFAULT_HELPER_MANAGER: true,
+  },
```
Comment (chancancode) [inline packages/@glimmer/integration-tests/lib/modes/env.ts:154]: With the feature being true by default we won't need this

**Review by chancancode — COMMENTED** [review]
Thanks for picking this up! Mostly looks good to me, just a few minor changes

**Comment by Windvis** [comment]
@chancancode Thanks for the quick review! I've pushed the requested changes :+1:.

**Review by chancancode — APPROVED** [review]

**Comment by chancancode** [comment]
Thank you! Released as 0.84.1!

**Pull request merged**